pkg/anchor/handler/proof/handler.go

```diff
@@ -91,9 +107,62 @@ func (h *WitnessProofHandler) HandleProof(witness *url.URL, anchorCredID string,
 		return fmt.Errorf("failed to add witness[%s] proof for credential[%s]: %w", witness.String(), anchorCredID, err)
 	}
 
-	vc.Proofs = append(vc.Proofs, witnessProof.Proof)
+	return h.handleWitnessPolicy(vc)
+}
 
-	h.vcCh <- vc
+func (h *WitnessProofHandler) handleWitnessPolicy(vc *verifiable.Credential) error {
+	witnessProofs, err := h.WitnessStore.Get(vc.ID)
+	if err != nil {
+		return fmt.Errorf("failed to get witness proofs for credential[%s]: %w", vc.ID, err)
+	}
+
+	ok, err := h.WitnessPolicy.Evaluate(witnessProofs)
+	if err != nil {
+		return fmt.Errorf("failed to evaluate witness policy for credential[%s]: %w", vc.ID, err)
+	}
+
+	if !ok {
```

Comment:
we also need to deal with the expiry of the offer period. we should have a policy to determine what to do.

Reply:
I already created issue #451